rfcs/2022-03-22-motion-rfc.md

```diff
@@ -0,0 +1,611 @@
+1. **Realtime**: User is directly interacting with an object on the UI.
+
+    *Example: When a user drags a carousel, the carousel slides as per the user's drag.*
+    
+    <img alt="Realtime Animation Example" src="./images/motion/realtime.gif" width="300px">
+
+2. **Non-Realtime**: The object's behavior is *post-interactive* i.e it occurs after a user's action, and is transitional.
+
+    *Example: When a user taps on a carousel card, it flips over within a set timeframe*
+    
+    <img alt="Realtime Animation Example" src="./images/motion/non-realtime.gif" width="300px">
+
+## Scope of Motion
+As of this RFC, we will be defining rules of motion only for **Non-Realtime** interactions. 
+
+Few examples of non-realtime motion are:
+1. Changing the size of a button on mouse hover
+2. Changing the opacity of a card on press
+3. Fading out a modal when the user closes it
+
+
+## Actions of Motion
+These are the actions an object on our UI can perform:
+
+1. **Scale**: An object can change it's dimensions (size) in X and Y axis
+
+    <img alt="scale" src="./images/motion/scale.gif" width="100px">
+
+3. **Move**: An object can change it's position in X and Y axis
+
+    <img alt="move" src="./images/motion/move.gif" width="100px">
+
+4. **Fade**: An object can change it's opacity within the range of 0 to 1
+
+    <img alt="fade" src="./images/motion/fade.gif" width="100px">
+
+5. **Fill**: An object can change it's color
+
+    <img alt="fill" src="./images/motion/fill.gif" width="100px">
+
+6. **Morph**: An object can morph itself into another object
+
+    <img alt="morph" src="./images/motion/morph.gif" width="100px">
+
+7. **Rotation**: An object can rotate itself in a range of 0 to 360 degrees
+
+    <img alt="rotate" src="./images/motion/rotate.gif" width="100px">
+
+## Rules (Tokens) of Motion
+While defining motion for an object transitioning from one state to another, we need to consider the following:
+1. **Delay**: When should the motion start?
+2. **Duration**: How long should an object take to transition from one state to another?
+3. **Easing**: How should an object accelerate/decelerate while transitioning from one state to another?
+
+### Delay
+You can *start* or *stop* your motion with some delay. For example, in the image below, it becomes difficult to reach a sub-menu in the drop down since the exit animation for the sub-menu starts instantly after the mouse is out of hover range. This can be fixed by adding a delay to the exit animation.
+> You can use `transition-delay` CSS property in the exit animation of the sub-menu to achieve this.
+
+<img alt="rotate" src="./images/motion/dropdown.gif" width="300px">
+
+#### Tokens - Delay
+We will be storing these tokens in `blade/src/tokens/global/motion.ts` as a `string` of milliseconds.
+
+> *Note: The naming for these tokens is not finalized yet. We will be updating this in the future.*
+
+Here is the **list of tokens** that we will store in Blade for `Delay`:
```

Comment:
How did we decide on these values or any rationale behind them?
PS: Can we link Dushyant's RFC/document here if any?

Reply:
I've added the link to his RFC in references